clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -141,4 +145,18 @@ public void run() {
         return file;
     }
 
+    /**
+     * Create a record buffer including the offset and message size at the start, which is required if the buffer is to
+     * be sent as part of `ProduceRequest`. This is the reason why we can't simply use
+     * `Record(long timestamp, byte[] key, byte[] value, CompressionType type, int valueOffset, int valueSize)` as this
+     * constructor does not include either of these fields.
+     */
+    public static ByteBuffer partitionRecordBuffer(long offset, Record record, CompressionType compressionType) {
```

Comment:
Yes, that's better. I'll have to add one additional case to the test to make sure multiple records work too.